src/main/java/se/sundsvall/oepintegrator/service/InstanceService.java

```diff
@@ -54,22 +54,14 @@ public void updateInstance(final String municipalityId, final String id, final I
 		}
 
 		instanceRepository.save(InstanceMapper.updateInstance(entity, instance, encryptedPassword));
-		clientFactory.removeClient(municipalityId, id);
+		clientFactory.removeClient(municipalityId, instance.getInstanceType());
 		clientFactory.createClient(entity);
 	}
 
 	public void deleteInstance(final String municipalityId, final String instanceId) {
```

Comment:
Transactional (to roll back deletion if clientFactory throws exception on removal)?